deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py

```diff
@@ -703,7 +1026,98 @@ def get_message(self, event_type=None):
+        self.hostinfo_basic_template = 'Host: <random_ip> (), open ports: '
+        self.protocols_list = ['udp', 'tcp']
+        self.localfile = '/var/log/nmap.log'
+
+    def generate_event(self):
+        number_open_ports = randint(1, 10)
+        host_ip = get_random_ip()
+        message_open_port_list = ''
+        for i in range(number_open_ports):
+            message_open_port_list += fr"{randint(1,65535)} ({choice(self.protocols_list)}) "
+
+        message = self.hostinfo_basic_template.replace('<random_ip>', host_ip)
+        message += message_open_port_list
+        message = fr"{self.HOSTINFO_MQ}:{self.localfile}:{message}"
+
+
+        return message
+
+
+class GeneratorWinevt:
+    """This class allows the generation of winevt events.
+
+    Create events of the different winevt channels: System, Security, Application, Windows-Defender and Sysmon.
+    It uses template events (`data/winevt.py`) for which the `EventID` field is randomized. Message structure:
+
+        f:EventChannel:{"Message":"<EVENTCHANNEL_MESSAGE>","Event":"<EVENT_CHANNEL_EVENT_XML>"}
+
+    Args:
+        agent_name (str): Name of the agent.
+        agent_id (str): ID of the agent.
+    """
+    def __init__(self, agent_name, agent_id):
+        self.agent_name = agent_name
+        self.agent_id = agent_id
+        self.WINENVT_MQ = 'f'
+        self.WINENVT = 'Eventchannel'
+        self.winevt_sources = [('system', winevt.WINEVT_SYSTEM), ('security', winevt.WINEVT_SECURITY),
+                               ('application', winevt.WINEVT_APPLICATION),
+                               ('windows-defender', winevt.WINEVT_WINDOWS_DEFENDER),
+                               ('sysmon', winevt.WINEVT_SYSMON)]
+
+        self.actual_winevt_source_index = -1
+
+    def generate_event(self, winevt_type=None):
+        """Genereate winevt event.
+
+        Generate the desired type of winevt event. If no type of winvt message is provided, all winvt message types
+        will be generated sequentially.
+
+        Args:
+            winevt_type (str): Winevt type message `system, security, application, windows-defender, sysmon`.
+        """
+        winevt_type_index = self.actual_winevt_source_index
+        if winevt_type is not None:
+            if winevt_type == 'system':
+                winevt_type_index = 0
+            elif winevt_type == 'security':
+                winevt_type_index = 1
+            elif winevt_type == 'application':
+                winevt_type_index = 2
+            elif winevt_type == 'windows-defender':
+                winevt_type_index = 3
+            elif winevt_type == 'sysmon':
+                winevt_type_index = 4
+        else:
+            self.actual_winevt_source_index = (self.actual_winevt_source_index + 1) % len(self.winevt_sources)
+
+        eventchannel_raw_message = self.winevt_sources[winevt_type_index][1]
+        eventchannel_raw_message = eventchannel_raw_message.replace("<random_int>", str(randint(0, 10*5)))
```

Comment:
By changing `winevt_sources` to a dictionary, this code is heavily simplified

Reply:
Yes. https://github.com/wazuh/wazuh-qa/commit/4614e3f1c54dfb77be82318af456b544785fe3a2